# AirBNB Listings Analysis for Boston and Seattle
Author: Thuy Trinh <br>
Date: 08 Oct 2021 <br>

**Objective**: 
1. Preprocessing data
2. Exploring dataset
3. Deliver insights

In [7]:
# import packages
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
# import data
# boston
calendar_b = pd.read_csv('data/raw/boston/calendar.csv')
listings_b = pd.read_csv('data/raw/boston/listings.csv')
reviews_b = pd.read_csv('data/raw/boston/reviews.csv')
# seattle
calendar_s = pd.read_csv('data/raw/seattle/calendar.csv')
listings_s = pd.read_csv('data/raw/seattle/listings.csv')
reviews_s = pd.read_csv('data/raw/seattle/reviews.csv')

## Question 1: General characteristics for rental market in the two cities:

### 1.1. What is the total number of rooms available for rent listed in AirBNB at the time of data collection? 

In [27]:
## For Boston
# find the time of data collection
collection_date_b = listings_b.last_scraped.unique()[0]
# find the total number of rooms avaiable
nb_rooms_b = len(listings_b.id.unique())

## For Seattle
# find the time of data collection
collection_date_s = listings_s.last_scraped.unique()[0]
# find the total number of rooms avaiable
nb_rooms_s = len(listings_s.id.unique())

## print results
print(f'As of {collection_date_b}, there are {nb_rooms_b} rooms in Boston listed for rent in AirBNB')
print(f'As of {collection_date_s}, there are {nb_rooms_s} rooms in Seattle listed for rent in AirBNB')

As of 2016-09-07, there are 3585 rooms in Boston listed for rent in AirBNB
As of 2016-01-04, there are 3818 rooms in Seattle listed for rent in AirBNB


## 1.2. What type of rooms are offered? Which types are most popular? What are the average rental rate for each type?

In [76]:
## For Boston
room_type_b_count = listings_b.room_type.value_counts()
room_type_b_perct = room_type_b_count/listings_b.shape[0]
room_type_b = pd.DataFrame([room_type_b_count, room_type_b_perct], index=['boston_count', 'boston_percentage']).transpose()
room_type_b

boston_count  boston_percentage
Entire home/apt        2127.0           0.593305
Private room           1378.0           0.384379
Shared room              80.0           0.022315

In [73]:
# compute average price per room type
# create new columns with `price` as numerical value by removing the $ sign
def convert_text_to_float(text):
    # removing the $ and , characters
    currency_char = '$,'
    for char in currency_char:
        text = text.replace(char, '')
    # return the text as float
    return float(text)
listings_b['price_numb'] = listings_b['price'].apply(convert_text_to_float)
room_type_price = listings_b[['room_type', 'price_numb', 'beds']].groupby(['room_type']).mean().round(0)
room_type_price

price_numb  beds
room_type                        
Entire home/apt       227.0   2.0
Private room           96.0   1.0
Shared room            93.0   1.0

In [77]:
room_type_b = room_type_b.join(room_type_price)
room_type_b

boston_count  boston_percentage  price_numb  beds
Entire home/apt        2127.0           0.593305       227.0   2.0
Private room           1378.0           0.384379        96.0   1.0
Shared room              80.0           0.022315        93.0   1.0

In [40]:
## For Seattle
room_type_s_count = listings_s.room_type.value_counts()
room_type_s_perct = room_type_s_count/listings_b.shape[0]
room_type_s = pd.DataFrame([room_type_s_count, room_type_s_perct], index=['seattle_count', 'seattle_percentage']).transpose()
room_type_s

seattle_count  seattle_percentage
Entire home/apt         2541.0            0.708787
Private room            1160.0            0.323570
Shared room              117.0            0.032636

In [78]:
listings_s['price_numb'] = listings_s['price'].apply(convert_text_to_float)
room_type_price = listings_s[['room_type', 'price_numb', 'beds']].groupby(['room_type']).mean().round(0)
room_type_s = room_type_s.join(room_type_price)
room_type_s

seattle_count  seattle_percentage  price_numb  beds
Entire home/apt         2541.0            0.708787       156.0   2.0
Private room            1160.0            0.323570        75.0   1.0
Shared room              117.0            0.032636        48.0   1.0

+ For both cities, there are only 3 types of rooms: `Entire home/apt`, `Private room` and `Shared room`. <br>
+ The majourity of the rentals are of the class `Entire home/apt` with around 60% of total rents in Boston and 71% of total rents in Seattle. <br>
+ In costrast, there are not a lot of `Shared room` available for rent in both cities, only 2% and 3% of total rents are share rooms in Boston and Seattle respectively.  
+ Rental price in Boston are much higher than in Seattle. An average house with 2 bedrooms in Boston will cost $227/night, while it's only $156/night in Seattle, which is 45% higher. Notablly, price of share room in Boston cost nearly twice as much of the cost in Seattle.    